sys/src/9/port/virtio_lib.c

```diff
@@ -26,7 +26,56 @@
 
 #include	"virtio_lib.h"
 
-#define MAXVQS 8 			// maximal number of VQs per device
+#define MAXVQS 8 			// maximal detectable number of VQs per device
+
+static uint32_t nvq;		// number of the detected virtio9p devices
+
+static Vqctl **cvq;			// array of device control structure pointers, length = nvq
+
+// Map device identifiers to descriptive strings to display in IO port
+// and interrupt allocation maps.
+
+typedef struct
+{
+	uint16_t did;
+	char *desc;
+} didmap;
+
+static didmap dmtab[] = {
```

Comment:
These numbers are in compact range, but do not start at 0 (0x1000 to 0x1009), not consecutive (RNG is 0x1005, 9P is 0x1009), and can be extended (e. g. there is virtio GPU which has its number, but it may be outside of this range). So instead of a loop (which runs once at device init) we need to add some range check etc. The present version with loop at least seems more open for future extensions.
Thanks.
PS CLANG failed me on a nested function, need to add a patch anyway.